[Ng Yong Xiang] iP

yongxiangng opened the pull request:
### Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,
* Text-based
* Easy to use
* Awesome like ~~me~~ you 😘 

All you need to do is,
1. Download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html) ⬇️ 
1. Double-click it 🖱️ 
1. Add your tasks 👍 
1. Let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (*coming soon*)
- [ ] Reminders (*coming soon*)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

What are you waiting for? Please review my PR and have mercy 🤔 🤔 🤔

Commit:
Add Gradle support

yongxiangng committed:
Add greet, echo, exit functionality to Duke bot

yongxiangng committed:
Add list and ability to print/add to list

yongxiangng committed:
Add Task Class and ability to mark task as done

yongxiangng committed:
Add Todos, Events and Deadlines as Tasks

yongxiangng committed:
Add automated text UI testing

yongxiangng committed:
Add exception handling for incorrect inputs by user

yongxiangng committed:
Revert "Add exception handling for incorrect inputs by user"

This reverts commit aee4bedc46f949fe9b1f3d62ecbe85562af8fba5.

yongxiangng committed:
Handle errors for incorrect user inputs

yongxiangng committed:
Update tests to expect invalid user inputs

yongxiangng committed:
Add support for deleting tasks from the list

yongxiangng committed:
Add expected error handling for deleting tasks

yongxiangng committed:
Use enum for command type in switch case when handling user input

yongxiangng committed:
Remove indentation for case clauses in switch statements

yongxiangng committed:
Add support to save task list to text file

yongxiangng committed:
Add support to enter task date in yyyy-mm-dd format

yongxiangng committed:
Merge branch 'branch-Level-7'

yongxiangng committed:
Merge branch 'branch-Level-8'

yongxiangng committed:
Refactor code into Ui, Storage, Parser and TaskList classes

yongxiangng committed:
Package code

yongxiangng committed:
Add JUnit tests

yongxiangng committed:
Update gitignore to ignore META-INF

yongxiangng committed:
Add JavaDocs

yongxiangng committed:
Refactor code to comply with coding standard

yongxiangng committed:
Add find command

yongxiangng committed:
Merge branch 'branch-A-JavaDoc'

yongxiangng committed:
Merge branch'branch-A-CodingStandard'

yongxiangng committed:
Merge branch 'branch-Level-9'

yongxiangng committed:
Merge branch 'add-gradle-support'

yongxiangng committed:
Use checkstyle

yongxiangng committed:
Fix style violation flagged by checkstyle

yongxiangng committed:
Add GUI for duke

yongxiangng committed:
Merge branch 'branch-Level-10'

yongxiangng committed:
Use varargs

HongshuW commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,106 @@
+            tasks1 = new TaskList();
+        }
+
+        tasks = tasks1;
```
Should the 'tasks1' variable be omitted? i.e. initialize 'tasks' directly.

HongshuW commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,8 @@
+ * Enum type of possible user commands.
+ */
+public enum Command {
+    LIST, DONE, TODO, DEADLINE, EVENT, DELETE, FIND, BYE
```
I like how you used enum here for commands. 👍

HongshuW commented on src/main/java/duke/io/Parser.java:
```diff
@@ -0,0 +1,187 @@
+     * @param input to be added and printed.
+     * @return string to inform user of successful command.
+     */
+    private String addTask(Task input) {
```
I like how you created a method for each command. This makes the code cleaner.

HongshuW commented on src/main/java/duke/io/Storage.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Loads task list from file path.
+     */
+    public ArrayList<Task> load() throws DukeException {
```
Should the javadoc for this method include @throws? I believe this would help other programmers to better understand this method.

HongshuW commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,113 @@
+     *
+     * @return the number of tasks in the list.
+     */
+    public int size() {
```
Perhaps you could use a verb for the name of this method, for example, getSize.

HongshuW reviewed (commented):
LGTM. Really neat code, also following the coding standard.

yongxiangng commented:
Thanks for the review & inputs @HongshuW 😀 You're right. I'll make the relevant changes 👍

Kangxun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,106 @@
+        Scanner scanner = new Scanner(System.in);
+        String input;
+
+        while (!(input = scanner.nextLine().trim()).equals("bye")) {
```
Perhaps a comment can be included to clarify what this condition checks for?

Ruofan1023 replied:
can consider abstract away the magic literal 'bye'

Kangxun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,106 @@
+    /**
+     * Runs the duke app.
+     */
+    public void run() {
```
i like how neat your run function is 👍🏻

Kangxun commented on src/main/java/duke/io/Parser.java:
```diff
@@ -0,0 +1,187 @@
+     *
+     * @param input command given by user input.
+     * @throws DukeException exception thrown due to invalid command or file writing error.
+     */
```
Should a `@return` be included here?

Kangxun commented on src/main/java/duke/io/Parser.java:
```diff
@@ -0,0 +1,187 @@
+     *
+     * @param userInput given by user.
+     * @return string to inform user of successful command.
+     */
```
Perhaps include a `@throws` javadoc here? It might apply to similar javadocs in this file as well.

Kangxun commented on src/test/java/task/DeadlineTest.java:
```diff
@@ -0,0 +1,27 @@
+
+import org.junit.jupiter.api.Test;
+
+import duke.task.Deadline;
```
Perhaps change the ordering of import statements for consistency? It might make it easier to browse the list.

Kangxun reviewed (commented):
Great work in following many of the best practices and coding standard guidelines! Only a few minor tweaks might be needed. Awesome work! 💯

yongxiangng committed:
Add assertions

Our code does not make use of assertions sufficiently.

Let's use assertions to make our code more easily maintainable.

This commit adds assertions to certain aspects of the code base which
currently assumes the correctness of the states.

It will be easier to uncover bugs if regressions happen.

yongxiangng committed:
Improve code quality

There is room for improvement in terms of code quality.

Let's refactor bits of our code and add header comments to increase the
quality of our code.

Improving our code quality makes our code more maintainable.

yongxiangng committed:
Merge pull request #1 from yongxiangng/branch-A-Assertions

Add assertions

yongxiangng committed:
Merge remote-tracking branch 'origin' into branch-A-CodeQuality

yongxiangng committed:
Merge pull request #2 from yongxiangng/branch-A-CodeQuality

Improve code quality

yongxiangng committed:
Add sort command

Users might want to use a sort command to sort their task list.

Let's implement a sort command to allow users to sort their task list.

yongxiangng committed:
Merge pull request #3 from yongxiangng/branch-C-Sort

Add sort command

yongxiangng committed:
Simplify iterations with stream for task list toString method

yongxiangng committed:
Merge pull request #4 from yongxiangng/branch-A-Streams

Simplify iterations with stream for task list toString method

yongxiangng committed:
Add CI (#5)

* Add CI

* Remove text redirection tests for JUnit testing

yongxiangng committed:
Add period tasks (#6)

* Add period tasks

* Fix checkstyle violations

yongxiangng committed:
Add support to add places to tasks

yongxiangng committed:
Merge pull request #7 from yongxiangng/branch-D-Places

Add support to add places to tasks

yongxiangng committed:
Fix mainClassName in build.gradle to build jar

yongxiangng committed:
Refactor Parser class for better abstractions

yongxiangng committed:
Anchor button and textfield to edges of window

yongxiangng committed:
Add user guide

yongxiangng committed:
Add release hyperlink to userguide

yongxiangng committed:
Remove header comments for main methods

yongxiangng committed:
Update README.md
